assert-plus/index.d.ts

```diff
@@ -3,65 +3,505 @@
 export function arrayOfFunc(options: any[], message ?: string): void;
+
+/**
+ * 
+ */
 export function arrayOfNumber(options: number[], message ?: string): void;
-export function arrayOfFinite(options: any[], message ?: string): void;
+
+/**
+ * 
+ */
+export function arrayOfFinite(options: number[], message ?: string): void;
+
+/**
+ * 
+ */
 export function arrayOfObject(options: any[], message ?: string): void;
+
+/**
+ * 
+ */
 export function arrayOfString(options: string[], message ?: string): void;
-export function arrayOfStream(options: any[], message ?: string): void;
-export function arrayOfDate(options: Array<string | number>, message ?: string): void;
-export function arrayOfRegexp(options: any[], message ?: string): void;
+
+/**
+ * 
+ */
+export function arrayOfStream(options: Stream[], message ?: string): void;
+
+/**
+ * 
+ */
+export function arrayOfDate(options: Date[], message ?: string): void;
+
+/**
+ * 
+ */
+export function arrayOfRegexp(options: RegExp[], message ?: string): void;
+
+/**
+ * 
+ */
 export function arrayOfUuid(options: string[], message ?: string): void;
-export function optionalArray(options: any, message ?: string): void;
-export function optionalBool(options: any, message ?: string): void;
-export function optionalBuffer(options: any, message ?: string): void;
-export function optionalFunc(options: any, message ?: string): void;
-export function optionalNumber(options: any, message ?: string): void;
-export function optionalFinite(options: any, message ?: string): void;
-export function optionalObject(options: any, message ?: string): void;
-export function optionalString(options: any, message ?: string): void;
-export function optionalStream(options: any, message ?: string): void;
-export function optionalDate(options: any, message ?: string): void;
-export function optionalRegexp(options: any, message ?: string): void;
-export function optionalUuid(options: any, message ?: string): void;
-export function optionalArrayOfArray(options: any, message ?: string): void;
-export function optionalArrayOfBool(options: any, message ?: string): void;
-export function optionalArrayOfBuffer(options: any, message ?: string): void;
-export function optionalArrayOfFunc(options: any, message ?: string): void;
-export function optionalArrayOfNumber(options: any, message ?: string): void;
-export function optionalArrayOfFinite(options: any, message ?: string): void;
-export function optionalArrayOfObject(options: any, message ?: string): void;
-export function optionalArrayOfString(options: any, message ?: string): void;
-export function optionalArrayOfStream(options: any, message ?: string): void;
-export function optionalArrayOfDate(options: any, message ?: string): void;
-export function optionalArrayOfRegexp(options: any, message ?: string): void;
-export function optionalArrayOfUuid(options: any, message ?: string): void;
+
+/**
+ * 
+ */
+export function optionalArray(options ?: any[], message ?: string): void;
```

Comment:
Yes, but `optionalArray()` will compile just fine, but it is pointless to call it with no arguments. `optionalArray(options: any[] | undefined, message?: string): void` will still allow to call `assert.optionalArray(undefined)` but not `assert.optionalArray()`.